### PR TITLE
[NSA-8536] Remove services and invites as part of user deactivation

### PR DESCRIPTION
**Pull request by insacuri**
Note: When testing this, do it on a test account as ticking the box and deactivating a user is quite destructive (removing all services and requests from the user)

### CODE REVIEW

File: src/app/users/postConfirmDeactivate.js [hunk 1]
```diff
@@ -42,6 +46,41 @@ const postConfirmDeactivate = async (req, res) => {
   } else {
     await deactivate(user.id, req.id);
     await updateUserIndex(user.id, req.id);
+    if(req.body["remove-services-and-requests"]) {
+      const userServiceRequests = await getUserServiceRequestsByUserId(user.id) || [];
+      for (let serviceRequest of userServiceRequests){
+        //Request status 0 is 'pending' and 2 is 'overdue'
+        if (serviceRequest.status === 0 || serviceRequest.status === 2) {
+          console.log("Rejecting service request with id: " + serviceRequest.id)
```
Comment (JaredAMarsh) [inline src/app/users/postConfirmDeactivate.js]: I would ideally replace this with `logger.debug` or `logger.info`, as there is no personal information in there, as console logs are fine but it'll go straight into app insights as well as the console with this 👍 And some of our apps don't load console logs into App Insights 😄 That and semi-colons, and spacing which ESLint should've highlighted but may not be working correctly...

Bit of a fussy thing I know but if I'm spotting small issues there's not much to worry about 😄

File: test/app/users/postConfirmDeactivate.test.js [hunk 2]
```diff
@@ -275,4 +326,57 @@ describe('When confirming deactivation of user given a reason from the select me
...
+      csrfToken: 'token',
+      backLink: 'services',
+      reason: '',
+      validationMessages: { reason: 'Please give a reason for deactivation' }
+    });
+    expect(res.redirect.mock.calls).toHaveLength(0);
+  });
+});
+
+describe('When the remove all services and requests checkbox is ticked', () => {
```
Comment (JaredAMarsh) [inline test/app/users/postConfirmDeactivate.test.js:345]: I'm happy with the current tests in this `describe` block, the additional things I'd like tested are:
- Checking the request filtering for statuses that are not `0` or `2`, for service requests at least.
- Checking that there is a call for each record of the for loops based on the access a user has e.g. 5 services are returned, 5 are removed and the same for requests 😄

I know those things are obvious looking at the code, but if we want tests to be a sort of living documentation then I think it's important these get captured for future us 👍

**Review by JaredAMarsh — CHANGES_REQUESTED** [review]
I've tested this in DEV and it worked as expected, a test user's services/roles were removed and all requests that were open or overdue were rejected 🧪👍 I don't see any issues with the updates from a logic standpoint 😄 

I have made some comments though, that whilst minor I think at least need to be addressed in a reply before approval, if not updated 👍

File: src/app/users/postConfirmDeactivate.js (hunk 1 above)
Comment (insacuri) [reply src/app/users/postConfirmDeactivate.js]: console.logs are now logger.info.  Also fixed the lint issues!

**Review by JaredAMarsh — APPROVED** [review]
Both the things I'd picked up previously have been addressed and the changes all make sense to me 👍

**Review by AMcCartneyMTH — APPROVED** [review]

**Pull request merged**